src/pad/src/RDLRouter.cpp

```diff
@@ -128,9 +129,138 @@ class RDLRouterGoalVisitor : public boost::default_astar_visitor
   }
 
  private:
-  RDLRouter::grid_vertex goal_;
+  grid_vertex goal_;
 };
 
+//////////////////////////////////////////////////////////////
+
+RDLRoute::RDLRoute(odb::dbITerm* source,
+                   const std::vector<odb::dbITerm*>& dests,
+                   const std::map<grid_vertex, odb::Point>& vertex_point_map)
+    : iterm_(source),
+      priority_(0),
+      vertex_point_map_(vertex_point_map),
+      terminals_(dests)
+{
+  terminals_.erase(std::remove_if(terminals_.begin(),
+                                  terminals_.end(),
+                                  [this](odb::dbITerm* other) {
+                                    return iterm_->getInst()
+                                           == other->getInst();
+                                  }),
+                   terminals_.end());
+
+  const odb::Point iterm_center = iterm_->getBBox().center();
+  std::stable_sort(terminals_.begin(),
+                   terminals_.end(),
+                   [&iterm_center](odb::dbITerm* lhs, odb::dbITerm* rhs) {
+                     const bool lhs_cover = RDLRouter::isCoverTerm(lhs);
+                     const bool rhs_cover = RDLRouter::isCoverTerm(rhs);
+                     // sort non-cover terms first
+                     if (lhs_cover == rhs_cover) {
+                       return odb::Point::squaredDistance(
+                                  iterm_center, lhs->getBBox().center())
+                              < odb::Point::squaredDistance(
+                                  iterm_center, rhs->getBBox().center());
+                     }
+                     return rhs_cover;
+                   });
+
+  resetRoute();
+}
+
+odb::dbITerm* RDLRoute::peakNextTerminal() const
+{
+  return *next_;
+}
+
+odb::dbITerm* RDLRoute::getNextTerminal()
+{
+  odb::dbITerm* iterm = peakNextTerminal();
+
+  next_++;
+
+  return iterm;
+}
+
+bool RDLRoute::compare(const std::shared_ptr<RDLRoute>& other) const
+{
+  const auto lhs_priority = -getPriority();
+  const auto rhs_priority = -other->getPriority();
```

Comment:
Priority is always first so why not just handle it up front
```
if (lhs_priority != rhs_priority) {
  return lhs_priority < rhs_priority ;
}
```
and let the rest of the code deal with distance alone if needed.